### PR TITLE
Add padding before first own message

### PR DESCRIPTION
**Pull request by p1gp1g**
## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Technical
- [ ] Other :

## Content

Add a padding before the first message send by ourselves. It is the same size than a displayed name.

## Motivation and context

It is easier to view our messages from others if the padding is a bit bigger.

## Screenshots / GIFs

|Before|After|
|-|-|
|Capture d’écran de 2022-03-01 00-39-42|Capture d’écran de 2022-03-01 00-35-07|

**Edit**: adding a table for better comparison of screnshoots

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s): 9

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [x] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [x] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR
- [x] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/item/AbsMessageItem.kt
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/room/detail/timeline/item/AbsMessageItem.kt:101]: Maybe using `isInvisible` like `holder.memberNameView.isInvisible = true` is enough, but this should also work

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/style/TimelineMessageLayout.kt
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/room/detail/timeline/style/TimelineMessageLayout.kt:27]: Nit: Maybe rename to `addTopMargin`

**Review by bmarty — COMMENTED** [review]
This looks good to me, thanks for the PR!
I ask for a design review from @amshakal 
Also can you add a small file for the changelog? You can see instruction [here](https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog)

Thanks!

**Comment by amshakal** [comment]
Hi there! Thanks for working on this. :) I agree with your solution. Before I approve, just want to check what's the padding (in pixels, if possible) you are using here?

**Comment by p1gp1g** [comment]
I'll do the variable rename & changelog file (.misc ?) when I've free time.

> Hi there! Thanks for working on this. :) I agree with your solution. Before I approve, just want to check what's the padding (in pixels, if possible) you are using here?

To have the same padding than the one before the bubble from other senders, I did an empty name instead of an explicit padding. That way, if you want to edit the name size, it will change that padding too.

**Review by bmarty — APPROVED** [review]
Thanks for the update!
I approve the PR, but @amshakal needs to approve it to so that we can merge it.

**Comment by p1gp1g** [comment]
> Be aware it will increase "blank" space if you have some decoration between the last from other and first from you like day separator or read receipts.

Indeed, here are some screenshots to illustrate : 
* Before, the space before our first message is the same than the space between 2 messages of the same sender. It might be preferred to save space since there is already some space between bubbles.
* After, the space before our first message is the same than before the first from someone else. It might be preferred to have the same padding every time it is a new sender. (I think it is easier to understand) *

| Before | After |
|-|-|
|Capture d’écran de 2022-03-01 23-58-49|Capture d’écran de 2022-03-01 23-45-45|
|Capture d’écran de 2022-03-01 23-59-24|Capture d’écran de 2022-03-01 23-45-04|
|Capture d’écran de 2022-03-02 00-04-08|image|

----

\* Comparison between same padding with reaction before/after
__before__ :

| 2 from the same sender | Our first message |
|-|-|
|Capture d’écran de 2022-03-02 00-21-06|image|

__after__ :

| Other's first message | Our first message |
|-|-|
|image|image|

**Comment by p1gp1g** [comment]
It is indeed better with a smaller padding. I've a bit edited how it works to use a 12dp space instead of the empty name.

Capture d’écran de 2022-03-02 17-53-08
Capture d’écran de 2022-03-02 17-53-33

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/item/AbsMessageItem.kt
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/room/detail/timeline/item/AbsMessageItem.kt:110]: Nit: it should be better to avoid using a negation here. So I would prefer to see
 
```kotlin
holder.additionalTopSpace.isVisible = attributes.informationData.messageLayout.addTopMargin
```

`isGone` -> `isVisible` and remove the `!`

Anyway, this is not a blocker.

**Review by bmarty — APPROVED** [review]
Thanks for the update. This PR will be merged if the CI is happy!

**Comment by bmarty** [comment]
Ah, I will need a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off) to be able to merge this PR.

@p1gp1g can you add a comment to this PR with your sign off please?

**Review by bmarty — CHANGES_REQUESTED** [review]
(waiting for sign off)

**Comment by p1gp1g** [comment]
I have removed the double negation and signed-off the commit

**Review by bmarty — APPROVED** [review]
Perfect! Thanks!

**Pull request merged**
